src/main/scala/zio/kafka/admin/AdminClient.scala

```diff
@@ -672,30 +692,63 @@ object AdminClient {
   )
 
   object TopicDescription {
-    def apply(jt: JTopicDescription): TopicDescription = {
+    def apply(jt: JTopicDescription): UIO[TopicDescription] = {
       val authorizedOperations = Option(jt.authorizedOperations).map(_.asScala.toSet)
-      TopicDescription(
-        jt.name,
-        jt.isInternal,
-        jt.partitions.asScala.toList.map(TopicPartitionInfo.apply),
-        authorizedOperations.map(_.map(AclOperation.apply))
-      )
+      UIO.foreach(jt.partitions.asScala.toList)(TopicPartitionInfo.apply).map { partitions =>
+        TopicDescription(
+          jt.name,
+          jt.isInternal,
+          partitions.toList,
+          authorizedOperations.map(_.map(AclOperation.apply))
+        )
+      }
     }
   }
 
-  case class TopicPartitionInfo(partition: Int, leader: Node, replicas: List[Node], isr: List[Node]) {
+  case class TopicPartitionInfo(partition: Int, leader: Option[Node], replicas: List[Node], isr: List[Node]) {
     lazy val asJava =
-      new JTopicPartitionInfo(partition, leader.asJava, replicas.map(_.asJava).asJava, isr.map(_.asJava).asJava)
+      new JTopicPartitionInfo(
+        partition,
+        leader.map(_.asJava).getOrElse(JNode.noNode()),
+        replicas.map(_.asJava).asJava,
+        isr.map(_.asJava).asJava
+      )
   }
 
   object TopicPartitionInfo {
-    def apply(jtpi: JTopicPartitionInfo): TopicPartitionInfo =
-      TopicPartitionInfo(
+    def apply(jtpi: JTopicPartitionInfo): UIO[TopicPartitionInfo] = {
+      val replicas = UIO.foreach(
+        jtpi
+          .replicas()
+          .asScala
+          .map(Node.apply)
+          .toList
+      ) {
+        case Some(node) => UIO.succeed(node)
+        case None       => ZIO.dieMessage("NoNode node not expected among topic replicas")
```

Comment:
I thjink we should be consistent in the failure mode (fail or die) between methods. 

Since the library always uses `Throwable` as its error channel, let's fail with `RuntimeException` as you did before. (using ZIO.getOrFailWith)